File: leetcode3/변지협/01_two_sum.py
'''
1. 아이디어 :
- 완전 탐색

2. 시간복잡도 :
    O(n^2)

3. 자료구조/알고리즘 :

'''


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(len(nums) - 1):
            for j in range(len(nums) - i - 1):
                sum = nums[i]
                sum += nums[i+j+1]
                if sum == target:
                    return [i,i+j+1]
            

File: leetcode3/이진희/01_Two Sum.java
/*

1. 아이디어 : 
    1차(투포인터): 2차원 배열로 (값, 인덱스) 함께 저장 후 정렬하여 탐색. O(NlogN) + O(N)
    2차(HashMap): 값:key, 인덱스:값 -> target을 만들기 위해 필요한 나머지 숫자 검색. O(N)

2. 시간복잡도 : O(N)

3. 자료구조/알고리즘 : HashMap

 */

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> map = new HashMap<>();
        
        for(int i=0; i<nums.length; i++) {
            int addNum = target-nums[i];

            if(map.containsKey(addNum)) return new int[]{map.get(addNum), i};

            map.put(nums[i], i);
        }
        return new int[]{-1,-1};
    }
}